app/models/miq_server/worker_management/monitor/kubernetes.rb

```diff
@@ -4,6 +4,11 @@ module MiqServer::WorkerManagement::Monitor::Kubernetes
   included do
     class_attribute :current_pods
     self.current_pods = Concurrent::Hash.new
+
+    class_attribute :current_deployments
+    self.current_deployments = Concurrent::Hash.new
+
+    attr_accessor :deployments_monitor_thread, :pods_monitor_thread
```

Comment:
Oh that's frustrating...I assumed we could watch both in the same thread...even so, 2 threads isn't that bad.

Reply:
Yeah, tell me about it.  Watches block until either the connection is severed or you bail out of it.  It's not even on the same endpoint.  pods are in the core api, and deployments are in the apps api.